site/app/controllers/forum/ForumController.php

```diff
@@ -916,6 +916,22 @@ public function getThreads($page_number = null) {
      * @Route("/courses/{_semester}/{_course}/forum/threads/single", methods={"POST"})
      */
     public function getSingleThread() {
+        // Checks if thread id is undefined. If so, render "fail" json response case informing that thread id is not set.
+        if (!isset($_POST['thread_id'])) {
+            return $this->core->getOutput()->renderJsonFail("Invalid thread id (UNSET ID)");
+        }
+        // Checks if thread id is null. If so, render "fail" json response case informing that thread id is null.
+        if ($_POST['thread_id'] == null) {
+            return $this->core->getOutput()->renderJsonFail("Invalid thread id (NULL ID)");
+        }
+        // Checks if thread id is not an integer value. If so, render "fail" json response case informing that thread id is not an integer value.
+        if (!(is_int($_POST['thread_id']) || ctype_digit($_POST['thread_id']))) {
```

Comment:
I'm torn with using only `is_int`.  I'm not exactly sure if it can be assume that `$_Post`  should only contain integers since strings composed of only numeric characters can currently be used interchangeably with their corresponding integer equivalent (i.e. using `"1"` vs `1`) when querying for a thread.